### ISSUE TITLE
Add Route Guard templates

### ISSUE THREAD

**Issue by doggy8088** [BeastCode/VSCode-Angular-TypeScript-Snippets#23]
There are four different route guard types we can use to protect our routes:

* CanActivate - Decides if a route can be activated
* CanActivateChild - Decides if children routes of a route can be activated
* CanDeactivate - Decides if a route can be deactivated
* CanLoad - Decides if a module can be loaded lazily

If you providing these route guard templates will be very nice.  

Thanks!

**Comment by BeastCode** [comment]
Great idea, thanks!

**Comment by BeastCode** [comment]
Do you have an example of what you think they should look like?

**Comment by doggy8088** [comment]
You can reference the default template in the Angular CLI `ng g guard` command.

Here is a sample that generated from `ng g guard login` command.  It's a `CanActivate` guard.

```ts
import { Injectable } from '@angular/core';
import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
import { Observable } from 'rxjs/Observable';

@Injectable()
export class LoginGuard implements CanActivate {
  canActivate(
    next: ActivatedRouteSnapshot,
    state: RouterStateSnapshot): Observable<boolean> | Promise<boolean> | boolean {
    return true;
  }
}
```

**Comment by BeastCode** [comment]
Thanks!!

**Comment by BeastCode** [comment]
Published in v4.2.1
